sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.aest.controller.getApplicationVersionHistory.e2e-spec.ts

```diff
@@ -0,0 +1,171 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../testHelpers";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { ApplicationStatus } from "@sims/sims-db";
+import { addDays, getPSTPDTDateHourMinute } from "@sims/utilities";
+
+describe("ApplicationAESTController(e2e)-getApplicationVersionHistory", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should get an array of application versions when there are an overwritten application and a completed application associated with the given application.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      submittedDate: new Date(),
+    });
+
+    const overwrittenApplication = await saveFakeApplication(
+      db.dataSource,
+      {
+        institution:
+          application.currentAssessment.offering.institutionLocation
+            .institution,
+        institutionLocation:
+          application.currentAssessment.offering.institutionLocation,
+        student: application.student,
+        program: application.currentAssessment.offering.educationProgram,
+        offering: application.currentAssessment.offering,
+        programYear: application.programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Overwritten,
+        applicationNumber: application.applicationNumber,
+        offeringIntensity:
+          application.currentAssessment.offering.offeringIntensity,
+        pirStatus: application.pirStatus,
+        submittedDate: addDays(-10),
+      },
+    );
+
+    const completedApplication = await saveFakeApplication(
+      db.dataSource,
+      {
+        institution:
+          application.currentAssessment.offering.institutionLocation
+            .institution,
+        institutionLocation:
+          application.currentAssessment.offering.institutionLocation,
+        student: application.student,
+        program: application.currentAssessment.offering.educationProgram,
+        offering: application.currentAssessment.offering,
+        programYear: application.programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Completed,
```

Comment:
Previous version of any given application cannot be in any status other than `Overwritten`. Hence either remove this version for test or set the status as overwritten.